python-backend/app/api/mines/mine/resources/mine.py

```diff
@@ -60,9 +62,41 @@ def get(self, mine_no_or_guid=None):
             # Handle MapView request
             _map = request.args.get('map', None, type=str)
             if _map and _map.lower() == 'true':
-                records = MineMapViewLocation.query.filter(MineMapViewLocation.latitude != None)
-                result = list((map(lambda x: x.json_for_map(), records)))
-                return {'mines': result}
+
+                # Below caches the mine map response object in redis with a timeout.
+                # Generating and jsonifying the map data takes 4-7 seconds with 50,000 points,
+                # so caching seems justified.
+                #
+                # TODO: Use some custom representation of this data vs JSON. The
+                # json string is massive (with 50,000 points: 16mb uncompressed, 2.5mb compressed).
+                # A quick test using delimented data brings this down to ~1mb compressed.
+                map_result = cache.get(MINE_MAP_CACHE)
+                last_modified = cache.get(MINE_MAP_CACHE + '_LAST_MODIFIED')
+                if not map_result:
+                    records = MineMapViewLocation.query.filter(MineMapViewLocation.latitude != None)
+                    last_modified = datetime.now()
+
+                    # jsonify then store in cache
+                    map_result = json.dumps(
+                        {
+                            'mines': list((map(lambda x: x.json_for_map(), records)))
+                        },
+                        separators=(',', ':'))
+                    cache.set(MINE_MAP_CACHE, map_result, timeout=43140)
```

Comment:
What's the reasoning behind this timeout value?

Reply:
I thought 12 hours would cover use during business hours, but ensure that when our ETLs run at night the cache will be expired the next day.

Reply:
and later on, we can use the scheduler to add that as well 👌